lib/flowViewport/sceneIndex/fvpPruningSceneIndex.cpp

```diff
@@ -0,0 +1,344 @@
+// Copyright 2024 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "fvpPruningSceneIndex.h"
+
+#include <pxr/base/tf/staticTokens.h>
+#include <pxr/imaging/hd/filteringSceneIndex.h>
+#include <pxr/imaging/hd/sceneIndexPrimView.h>
+#include <pxr/imaging/hd/tokens.h>
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+TF_DEFINE_PUBLIC_TOKENS(FvpPruningTokens, FVP_PRUNING_TOKENS);
+
+PXR_NAMESPACE_CLOSE_SCOPE
+
+PXR_NAMESPACE_USING_DIRECTIVE
+
+namespace {
+
+template<typename Container>
+bool _HasAncestorInclusiveInContainer(const SdfPath& path, const Container& pathsContainer) {
+    SdfPath currPath = path;
+    while (!currPath.IsEmpty() && !currPath.IsAbsoluteRootPath()) {
+        if (pathsContainer.find(currPath) != pathsContainer.end()) {
+            return true;
+        } else {
+            currPath = currPath.GetParentPath();
+        }
+    }
+    return false;
+}
+
+bool _MeshesFilterHandler(const HdSceneIndexBaseRefPtr& sceneIndex, const SdfPath& primPath, const HdSceneIndexPrim& prim)
+{
+    // Currently we just flat out remove any prim with a mesh type. If we were to add extra checks to make sure this is not
+    // a mesh prim that serves another purpose, we would add them here.
+    return prim.primType == HdPrimTypeTokens->mesh;
+}
+
+bool _CapsulesFilterHandler(const HdSceneIndexBaseRefPtr& sceneIndex, const SdfPath& primPath, const HdSceneIndexPrim& prim)
```

Comment:
Minor: isn't it redundant to pass in both the prim and the primPath?  Since you're passing in the sceneIndex, you can derive the prim from the primPath.

Reply:
Yes, but since the calling code has already retrieved the prim anyways, not passing it would mean the filters would have to get the prim again, so it'd be redundant in that way (and depending on the upstream scene indices, might incur a performance cost?). And since I feel most filters would end up getting the prim anyways since you can't do much only off a path, it feels simpler to me to just pass it from the get-go.